### PR TITLE
Added support for protocol buffers and integrations test added with refactors

### PR DESCRIPTION
**Pull request by Raalsky**

### CODE REVIEW

File: tests/integration/test_read_only_run.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+from neptune_fetcher import (
+    ReadOnlyProject,
+    ReadOnlyRun,
+)
+
+
+@pytest.fixture(scope="session")
+def api_token():
+    return base64.b64encode(json.dumps({"api_address": ""}).encode()).decode()
```
Comment (AleksanderWWW) [inline tests/integration/test_read_only_run.py:44]: This is duplicated in another test file. Maybe a `conftest.py` will be in order? (plus a type hint for return type)

File: tests/integration/test_read_only_run.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+            FloatField(path="metrics/float", value=25.97),
+            IntField(path="metrics/int", value=97),
+            StringField(path="metrics/string", value="Test string"),
+        ]
+
+
+@pytest.fixture(scope="session")
+def hosted_backend() -> HostedNeptuneBackend:
+    with patch("neptune_fetcher.read_only_project.HostedNeptuneBackend", BackendMock) as mock:
+        yield mock
```
Comment (AleksanderWWW) [inline tests/integration/test_read_only_run.py:115]: Same here. `conftest.py`

File: tests/integration/test_read_only_run.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+    ReadOnlyRun,
+)
+
+
+@pytest.fixture(scope="session")
+def api_token():
+    return base64.b64encode(json.dumps({"api_address": ""}).encode()).decode()
+
+
+class BackendMock:
```
Comment (AleksanderWWW) [inline tests/integration/test_read_only_run.py:47]: Is it possible to reuse the `BackendMock` from the other module?

Comment (Raalsky) [reply tests/integration/test_read_only_run.py]: Done, moved to `conftest.py`.

File: tests/integration/test_read_only_run.py (hunk 1 above)
Comment (Raalsky) [reply tests/integration/test_read_only_run.py]: Done

**Review by AleksanderWWW — APPROVED** [review]

**Review by AleksanderWWW — APPROVED** [review]

**Pull request merged**